### PR TITLE
Cuffable/Handcuff ECS

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

- predicted
- ECS
- cleanup
- makes the brain tingle
- i tested it bro trust me

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no cl no fun

### CODE REVIEW

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Access attribute + VVAccess.ReadWrite (gottem)

File: Content.Shared/Cuffs/SharedCuffableSystem.cs [hunk 1]
```diff
@@ -74,54 +168,478 @@ public void UpdateCuffState(EntityUid uid, SharedCuffableComponent component)
...
+                return;
+
+            if (args.Handled)
+                return;
+            args.Handled = true;
+            component.Cuffing = false;
+
+            if (_net.IsServer)
+            {
+                // TODO these pop-ups need third-person variants (i.e. {$user} is cuffing {$target}!
```
Comment (AJCM-git) [inline Content.Shared/Cuffs/SharedCuffableSystem.cs:288]: do this

File: Content.Shared/Cuffs/SharedCuffableSystem.cs [hunk 2]
```diff
@@ -74,54 +168,478 @@ public void UpdateCuffState(EntityUid uid, SharedCuffableComponent component)
...
+            {
+                if (_net.IsServer)
+                {
+                    _popup.PopupEntity(Loc.GetString("handcuff-component-target-has-no-free-hands-error",
+                        ("targetName", Identity.Name(target, EntityManager, user))), user, user);
+                }
+                return;
+            }
+
+            // TODO these messages really need third-party variants. I.e., "{$user} starts cuffing {$target}!"
```
Comment (AJCM-git) [inline Content.Shared/Cuffs/SharedCuffableSystem.cs:445]: same

**Review by AJCM-git — CHANGES_REQUESTED** [review]
just some details (formatting and nesting in SharedCuffableSystem.cs look a lil uggo sometimes tho but its fine)

**Comment by AJCM-git** [comment]
merge master

**Comment by AJCM-git** [comment]
linter error

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**